openstates/fl/bills.py

```diff
@@ -268,24 +250,179 @@ def scrape_vote(self, bill, chamber, date, url):
             else:
                 break
 
-        totals = re.search(r'(?ms)\s+(\d+)\s+(\d+)\s+.*?TOTALS', text).groups()
+        totals = re.search(
+            r'(?msu)\s+(\d{1,3})\s+(\d{1,3})\s+.*?TOTALS', text).groups()
         yes_count = int(totals[0])
         no_count = int(totals[1])
-        if yes_count != len(votes['yes']) or no_count != len(votes['no']):
-            raise AssertionError(
-                    "Should have {0} yes votes: {1}\n".
-                    format(yes_count, votes['yes']) +
-                    "and {0} no votes: {1}".
-                    format(no_count, votes['no'])
-                    )
         passed = (yes_count > no_count)
         other_count = len(votes['other'])
 
-        vote = Vote(chamber, date, motion, passed, yes_count, no_count,
+        vote = Vote('upper', date, motion, passed, yes_count, no_count,
                     other_count)
         vote.add_source(url)
         vote['yes_votes'] = votes['yes']
         vote['no_votes'] = votes['no']
         vote['other_votes'] = votes['other']
 
+        vote.validate()
+        bill.add_vote(vote)
+
+    def scrape_floor_vote(self, chamber, bill, date, url):
+        (path, resp) = self.urlretrieve(url)
+        text = convert_pdf(path, 'text')
+        lines = text.split("\n")
+        os.remove(path)
+
+        MOTION_INDEX = 4
+        TOTALS_INDEX = 6
+        VOTE_START_INDEX = 9
+
+        motion = lines[MOTION_INDEX].strip()
+        assert motion, "Senate floor vote's motion name appears to be empty"
+        if lines[MOTION_INDEX + 1].strip():
+            motion = "{}, {}".format(motion, lines[5].strip())
+            TOTALS_INDEX += 1
+            VOTE_START_INDEX += 1
+
+        (yes_count, no_count, other_count) = [int(x) for x in re.search(
+            r'^\s+Yeas - (\d+)\s+Nays - (\d+)\s+Not Voting - (\d+)\s*$',
+            lines[TOTALS_INDEX]).groups()]
```

Comment:
I ran this successfully on the current session, but I don't think there are any floor votes available. When I ran it on 2014, however, I got the following error on this line. Is there something else I was supposed to be checking?

11:59:36 INFO scrapelib: GET - http://flsenate.gov/Session/Bill/2014/0089/Vote/HouseVote_h0089e1483.PDF
Traceback (most recent call last):
  File "/home/rachel/.virtualenvs/openstates/bin/billy-update", line 9, in <module>
    load_entry_point('billy==1.8.0', 'console_scripts', 'billy-update')()
  File "/home/rachel/.virtualenvs/openstates/local/lib/python2.7/site-packages/billy/bin/update.py", line 392, in main
    run_record += _run_scraper(stype, args, metadata)
  File "/home/rachel/.virtualenvs/openstates/local/lib/python2.7/site-packages/billy/bin/update.py", line 103, in _run_scraper
    scraper.scrape(time, chambers=chambers)
  File "/home/rachel/work/openstates/openstates/fl/bills.py", line 36, in scrape
    bill_id, title, sponsor, url)
  File "/home/rachel/work/openstates/openstates/fl/bills.py", line 193, in scrape_bill
    self.scrape_floor_vote('lower', bill, vote_date, vote_url)
  File "/home/rachel/work/openstates/openstates/fl/bills.py", line 289, in scrape_floor_vote
    lines[TOTALS_INDEX]).groups()]
AttributeError: 'NoneType' object has no attribute 'groups'